website/src/pages/download.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: Download
```

Comment:
This page should only provide ASF release download. Aka, the link to https://dlcdn.apache.org/incubator/opendal/0.36.0/.

All other artifact like rust, java are not part of ASF release, and should not be listed here.